pkg/kritis/metadata/metadata.go

```diff
@@ -16,13 +16,32 @@ limitations under the License.
 
 package metadata
 
+import (
+	kritisv1beta1 "github.com/grafeas/kritis/pkg/kritis/apis/kritis/v1beta1"
+	"github.com/grafeas/kritis/pkg/kritis/secrets"
+	containeranalysispb "google.golang.org/genproto/googleapis/devtools/containeranalysis/v1alpha1"
+)
+
 type MetadataFetcher interface {
-	// Get Package Vulnerabilites
+	// Get Package Vulnerabilites for a given image.
 	GetVulnerabilities(containerImage string) ([]Vulnerability, error)
+	// Create Attesatation Occurrence for an image.
+	CreateAttestationOccurence(note *containeranalysispb.Note,
+		containerImage string,
+		pgpSigningKey *secrets.PgpSigningSecret) (*containeranalysispb.Occurrence, error)
+	// Get Attestation Note for an Attestation Authority.
+	GetAttestationNote(aa kritisv1beta1.AttestationAuthority) (*containeranalysispb.Note, error)
+	// Get Attestation Occurrences for given image.
+	GetAttestations(containerImage string) ([]PgpAttestation, error)
 }
 
 type Vulnerability struct {
 	Severity        string
 	HasFixAvailable bool
 	CVE             string
 }
+
+type PgpAttestation struct {
```

Comment:
PGPAttestation? Words in names that are initialisms or acronyms (e.g. "URL" or "NATO") have a consistent case. For example, "URL" should appear as "URL" or "url" (as in "urlPony", or "URLPony"), never as "Url".

Reply:
Also, since this is a public struct, should it get documentation?

Reply:
we haven't documented most of the public structs. I will add one for this one and probably we can go back add later for other structs.